packages/network/src/connection/ws/ServerWsEndpoint.ts

```diff
@@ -177,9 +189,42 @@ export async function startHttpServer(
     } else {
         throw new Error('must supply both privateKeyFileName and certFileName or neither')
     }
-
-    httpServer.listen(port)
-    await once(httpServer, 'listening')
-    staticLogger.trace(`started on port %s`, port)
+    // clean up Unix Socket
+    if (typeof config === "string") {
+        httpServer.on('error', (err) => {
+            if (err.message.includes('EADDRINUSE')) {
+                staticLogger.info("socket already in use, checking if it is used by another server")
+                const clientSocket = new net.Socket()
+                clientSocket.on('error', function(err) {
+                    if (err.message.includes('ECONNREFUSED')) {  // No other server listening
+                        staticLogger.info("cleaning unused socket")
+                        fs.unlinkSync(config)
+                        httpServer.listen(config, function() {
+                            staticLogger.info('server recovered')
+                            return httpServer
+                        })
+                    }
+                })
+                clientSocket.connect({path: config},function() {
+                    staticLogger.trace('Server is already running in ' + config)
+                    process.exit()
+                })
+            }
+        })
+    }
+    try {
+        httpServer.listen(config)
+        await once(httpServer, 'listening')
+        staticLogger.info(`listening on %s`, JSON.stringify(config))
+    } catch (err) {
```

Comment:
What throws? the `once`?

Reply:
Seems that the await once() throws here when the httpServer errors with EADDRINUSE